lib/DIKUrevy/Website/Controller/Auth.pm

```diff
@@ -84,6 +88,77 @@ sub create_user_submit {
     return $self->redirect_to('frontpage');
 }
 
+sub random_password() {
+    my @chars = ('A'..'Z', 'a'..'z', '0'..'9', '@%$#&+*');
+    return join('', map { $chars[rand @chars] } 1..16);
+}
+
+sub new_password {
+    my $self = shift;
+
+    my $jwt = $self->req->query_params->param('jwt');
+    if ($jwt) {
+        my $secret = $self->config('secrets')->[0];
+        my $claims;
+        eval { $claims = Mojo::JWT->new(secret => $secret)->decode($jwt); };
+        if ((! $claims) || (time() > $claims->{'exp'})) {
+            $self->show_error('Ugyldigt løsen-link.', flash => 1);
+            return $self->redirect_to('frontpage');
+        }
+        else {
+            my $user = DIKUrevy::User->retrieve( { id => $claims->{'id'} } );
+            my $new_password = random_password();
+            $user->set_password($new_password);
+            $user->save;
+            $self->show_message('Du kan nu logge ind med dit nye løsen: '
+                                . $new_password,
+                                flash => 1);
```

Comment:
Hmm. I stedet for at sætte et nyt løsen, så kunne vi kalde

```
$self->authenticate(undef, undef, { auto_validate => $claims->{id} });
```
og så redirecte til bruger-redigerings-siden?

Så vidt jeg kan læse på [dokumentationen for Mojolicious::Plugin::Authentication](http://search.cpan.org/~madcat/Mojolicious-Plugin-Authentication-1.26/lib/Mojolicious/Plugin/Authentication.pm#authenticate($username,_$password,_$extra_data_hashref)), så burde dét sætte en login-cookie og gå udenom bruger-checket.